api/app/src/command/connected-user/get-connected-user.ts

```diff
@@ -34,10 +34,14 @@ export const getProviderDataFromConnectUserOrFail = (
   provider: ProviderOptions
 ) => {
   if (!connectedUser.providerMap) throw new NotFoundError("Could not find provider map");
-  const providerData = connectedUser.providerMap[provider];
-  if (!providerData) throw new NotFoundError("Provider map has no data");
 
-  return providerData;
+  if (connectedUser.providerMap[provider]?.token) {
+    const providerData = connectedUser.providerMap[provider];
+    if (!providerData) throw new NotFoundError("Provider map has no data");
+    return providerData;
```

Comment:
I don't think we can wrap the original check inside the check for token... like 40 would never trigger, since it can only execute if there's something on `providerData` (at least the token), right?

Reply:
Changed the logic of the function!